core/kernel.rbs

```diff
@@ -614,7 +614,15 @@ module Kernel : BasicObject
   #
   # See also String#to_r.
   #
-  def self?.Rational: (Numeric | String | Object x, ?Numeric | String y, ?exception: bool exception) -> Rational
+  def self?.Rational: (_ToInt | _ToR numer, ?_ToInt | _ToR denom, exception: false) -> Rational?
+                    | (_ToInt | _ToR numer, ?_ToInt | _ToR denom, ?exception: bool) -> Rational
```

Comment:
With `exception: false` already covered, can this simply be
```suggestion
                    | (_ToInt | _ToR numer, ?_ToInt | _ToR denom, ?exception: true) -> Rational
```
?

Also applies to other conversion methods in Kernel.
Does not apply to the `[T]` ones if they don’t squelch exceptions as you said.

BTW,
```ruby
Rational 1, 0, exception: false #=! ZeroDivisionError: divided by 0
```

Reply:
the `: false` case is for when a typechecker can explicitly determine that it's falsey. the `: bool` variant is when the typechecker doesn't have enough information.

Reply:
It might make sense to "default" to having `Rational?` be the default

Reply:
> the `: false` case is for when a typechecker can explicitly determine that it's falsey. the `: bool` variant is when the typechecker doesn't have enough information.

Not falsey, just `false`. Ruby rejects `exception: nil`. It also rejects `boolish`es, only accepting `true`.

Reply:
Could there be a world where `bool` has three values? 💭